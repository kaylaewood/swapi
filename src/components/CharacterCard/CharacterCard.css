.character-card {
  background-color: #0B1E36;
  box-shadow: 15px 15px 15px #A6A6A6;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  margin: 2vw;
  padding: 2vw;
  text-align: center;
}

h4 {
  padding-top: 2vw;
}

.favorite-button {
  font-size: 1vw;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: .5vw;
  width: 15vw;
}
