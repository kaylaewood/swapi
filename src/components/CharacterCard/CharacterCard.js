import React, { Component } from 'react';
import './CharacterCard.css';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

function CharacterCard({ character }) {
  const showFilms = () => {
    let counter=0;
    return character.films.map(film => {
      counter++;
      return <p key={counter}>- {film}</p>
    })
  }

  return (
    <section className='character-card'>
      <h2>{character.name}</h2>
      <h4>SPECIES: <br /> {character.species}</h4>
      <h4>HOMEWORLD: <br /> {character.homeworld[0]}</h4>
      <h4>POPULATION: <br /> {character.homeworld[1]}</h4>
      <h4>FILMS:</h4>
      {showFilms()}
    </section>
  )
}

CharacterCard.propTypes = {
  character: PropTypes.object
}

export default CharacterCard;
