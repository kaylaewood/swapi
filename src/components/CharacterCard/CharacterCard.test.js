import React from 'react';
import ReactDOM from 'react-dom';
import CharacterCard from './CharacterCard';
import { shallow } from 'enzyme';

describe('CharacterCard', () => {
  let character = {
    name: 'Bob',
    homeworld: ['Earth', 7000000000],
    species: 'human',
    films: ['Pulp Fiction', 'Empire Strikes Back', 'Casablanca']
  }
  it('should match the snapshot', () => {
    const wrapper = shallow(<CharacterCard
      character={character}
    />)
    expect(wrapper).toMatchSnapshot();
    })
})
