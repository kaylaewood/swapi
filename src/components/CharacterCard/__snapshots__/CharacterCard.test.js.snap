// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should match the snapshot 1`] = `
<section
  className="character-card"
>
  <h2>
    Bob
  </h2>
  <h4>
    SPECIES: 
    <br />
     
    human
  </h4>
  <h4>
    HOMEWORLD: 
    <br />
     
    Earth
  </h4>
  <h4>
    POPULATION: 
    <br />
     
    7000000000
  </h4>
  <h4>
    FILMS:
  </h4>
  <p
    key="1"
  >
    - 
    Pulp Fiction
  </p>
  <p
    key="2"
  >
    - 
    Empire Strikes Back
  </p>
  <p
    key="3"
  >
    - 
    Casablanca
  </p>
</section>
`;
